### ISSUE TITLE
Multiple e.g. Features Start/Finish like in SourceTree

### ISSUE THREAD

**Issue by darkman1983** [OpherV/gitflow4idea#189]
Hi,
personally it's a real pain, if you could only start one e.g. feature.
Because things can change like if you have to get some more information to get your feature finished and your supervisor is currently not available.
So you would normally freeze the coding on that feature and start to work on another one until you get the information needed to finish that feature.

In SourceTree it looks like this (sorry for german Sreenshots, can't seem to get SourceTree back to english):

This is the normal
![SourceTree - Start|Stop](https://thumb.ibb.co/iMa2aS/2018_03_26_10_20_05_Sourcetree.jpg)
The view like in this plugin, except the "More Action" button.

![SourceTree - More Actions menu](https://thumb.ibb.co/cKh4pn/2018_03_26_10_21_19.jpg)
So there you have
- Finish feature
- Finish version
- Finish Hotpatch

![SourceTree - More Actions menu - Already started a e.g. feature](https://thumb.ibb.co/dNyNaS/2018_03_26_10_22_45.jpg)
So in this case
- Start new feature
- Finish feature
- Start new version
- Finish version
- Start new hotpatch
- Finish hotpatch

And Finally e.g. "Finish Feature"...
![SourceTree - Dialog - Select feature to finish](https://thumb.ibb.co/juX4pn/2018_03_26_10_23_05_Sourcetree.jpg)
And there you have it, select which feature you want to finish.

I think this should be implemented, as it's very needed.

Greetings
Dark

**Comment by OpherV** [comment]
I apologize - but I don't understand the feature request. Can you please elaborate?

Are you talking about finishing multiple features at the same time? How often does that happen (and why?). How come you're finishing a feature you're not actively working on?

**Comment by darkman1983** [comment]
Yes i do.

Web use Scrum, a sprint is 2 weeks for us. In this time it could happen 2-3 times, that you have to start another feature, because of you can't finish the current.
This could happen e.g. if your Supervisor is in holiday and the deputy has no permission to take decisions.

So again you have to start multiple features at same time and finish another one later.

**Comment by OpherV** [comment]
I'm afraid I still don't quite grasp what you're suggesting.
What changes to the UI do you envision?

**Comment by unknownUser12345** [comment]
Hi, 
as a colleague of darkman1983 I will try to clarify his request.
As he mentioned, every developer is working on several tasks during a scrum sprint.
So it might happen that an unsolvable problem occurs, which can only be solved by a decision of a third involved person (e.g. the product owner).
As this person is not always available and the developer should be able to continue his work (working time is money), it is
sometimes neccesary that he switches the task.
So 'task a' has not been finished (-> paused) and 'task b' is the new task the user is working on.

I have to admit, that I currently dont know the UI of your implementation, because
it was darkman1983's task to find a way to integrate gitflow into intellij. 
And - surprise, surprise - darkman1983 is currently not available due to health reasons...

Indeed it is not very helpful to post screenshots in german language as they are almost worthless
for someone who does not speak german -.- sorry about that.

As I understand his post, he tried to compare the functionality of SourceTree to your implementation.
More precisely, he asked if it is possible to start a "task" (I dont know the exact wording, sorry), pause it sometimes
and start a second "task" in parallel. So the user is not forced to finish 'task a' before starting 'task b' and is able to switch
between them as it might be neccessary.

I hope this explanation will help a little to resolve the confusion (did it ?).

Thanks in advance for taking your time :) !

Greetings,
Niko

**Comment by OpherV** [comment]
The functionality you described already exists...

You can always start a feature, then a start a new feature without finishing the first one and switch between features. 

![gitflow1](https://user-images.githubusercontent.com/3951311/37965429-c696c7ce-31cd-11e8-9ea4-560aa4f10164.gif)

You can even do the same with the intelliJ task interface.

![gitflow2](https://user-images.githubusercontent.com/3951311/37965435-cbcbdb58-31cd-11e8-8f6a-c2709f872ea1.gif)

**Comment by unknownUser12345** [comment]
Ok... thats pretty nice :). 
Thanks for your detailled explanation and sorry for the time you wasted on thinking about the meaning of the request.

Kind regards,
Niko
